subsys/bluetooth/host/iso.c

```diff
@@ -368,10 +368,10 @@ void bt_iso_connected(struct bt_conn *iso)
 
 	err = bt_iso_setup_data_path(chan);
 	if (err != 0) {
-		LOG_ERR("Unable to setup data path: %d", err);
+		if (false) {
 #if defined(CONFIG_BT_ISO_BROADCAST)
-		if (iso->iso.info.type == BT_ISO_CHAN_TYPE_BROADCASTER ||
-		    iso->iso.info.type == BT_ISO_CHAN_TYPE_SYNC_RECEIVER) {
+		} else if (iso->iso.info.type == BT_ISO_CHAN_TYPE_BROADCASTER ||
+			   iso->iso.info.type == BT_ISO_CHAN_TYPE_SYNC_RECEIVER) {
```

Comment:
It's really hard to read and understand a code chopped like this.
I guess the only thing you need to fix this is to add `return` statement after this:

```
			err = bt_iso_big_terminate(big);
			if (err != 0) {
				LOG_ERR("Could not terminate BIG: %d", err);
			}
			return;   // <= HERE
```

Reply:
That is certainly an option, but then we would end up with a code block that looks like

```c
if (broadcast) {
    // handle broadcast stuff
    return;
} else if (unicast) {
    // handle unicast stuff
} else {
    __ASSERT(...)
}

return; 
```

Which also is somewhat funky. 

The `if (false)` approach is something we use quite often in Zephyr (or at least in the BT subsystem) for this specific sort of thing. 

What about this alternative, where I change the order of unicast and broadcast to avoid the `if (false)`? 

```c
		if (IS_ENABLED(CONFIG_BT_ISO_UNICAST) &&
		    iso->iso.info.type == BT_ISO_CHAN_TYPE_CONNECTED) {
			bt_conn_disconnect(iso,
					   BT_HCI_ERR_REMOTE_USER_TERM_CONN);
#if defined(CONFIG_BT_ISO_BROADCAST)
		} else if (iso->iso.info.type == BT_ISO_CHAN_TYPE_BROADCASTER ||
			   iso->iso.info.type == BT_ISO_CHAN_TYPE_SYNC_RECEIVER) {
			struct bt_iso_big *big;
			int err;

			big = lookup_big_by_handle(iso->iso.big_handle);

			err = bt_iso_big_terminate(big);
			if (err != 0) {
				LOG_ERR("Could not terminate BIG: %d", err);
			}
#endif /* CONFIG_BT_ISO_BROADCAST */
		} else {
			__ASSERT(false, "Invalid iso.info.type: %u",
				 iso->iso.info.type);
		}
```

Reply:
I meant something like:
```
if (broadcast) {
    // handle broadcast stuff
    return;
}

if (unicast) {
    // handle unicast stuff
} else {
    __ASSERT(...)
}

return; 
```

Reply:
> I meant something like:
> 
> ```
> if (broadcast) {
>     // handle broadcast stuff
>     return;
> }
> 
> if (unicast) {
>     // handle unicast stuff
> } else {
>     __ASSERT(...)
> }
> 
> return; 
> ```
> 
> so the fix would look like this:
> 
> ```
> diff --git a/subsys/bluetooth/host/iso.c b/subsys/bluetooth/host/iso.c
> index c3ff5ee9df..448f3f3dbb 100644
> --- a/subsys/bluetooth/host/iso.c
> +++ b/subsys/bluetooth/host/iso.c
> @@ -381,6 +381,8 @@ void bt_iso_connected(struct bt_conn *iso)
>                         if (err != 0) {
>                                 LOG_ERR("Could not terminate BIG: %d", err);
>                         }
> +
> +                       return;
>                 }
>  #endif /* CONFIG_BT_ISO_BROADCAST */
>                 if (IS_ENABLED(CONFIG_BT_ISO_UNICAST) &&
> ```

I am aware, which is the semantic equivalent of 

```
if (broadcast) {
    // handle broadcast stuff
    return;
} else if (unicast) {
    // handle unicast stuff
} else {
    __ASSERT(...)
}

return;
```

My point being is that it is somewhat odd to insert a `return` in just one of the `if` clauses.

Reply:
I was one who use the `if (0)` or `if (false)` extensively in Controller code where multiple exclusive, or combinations,  or all inclusive control path be conditionally compiled. Using the `if (false)` gave a style that was scalable in terms of number of build combinations.
Given sufficient newlines between the blocks, the code can be sufficiently readable.